frontend/src/pages/ExperimentDetails.test.tsx

```diff
@@ -0,0 +1,453 @@
+
+    expect(tree.state('activeRecurringRunsCount')).toBe(2);
+    tree.unmount();
+  });
+
+  it('opens the recurring run manager modal when \'manage\' is clicked', async () => {
+    await mockNJobs(1);
+    const tree = TestUtils.mountWithRouter(<ExperimentDetails {...generateProps() as any} />);
+    await TestUtils.flushPromises();
+
+    tree.update();
+
+    tree.find('#manageExperimentRecurringRunsBtn').at(0).simulate('click');
+    await TestUtils.flushPromises();
+    expect(tree.state('recurringRunsManagerOpen')).toBe(true);
+    tree.unmount();
+  });
+
+  it('closes the recurring run manager modal', async () => {
+    await mockNJobs(1);
+    const tree = TestUtils.mountWithRouter(<ExperimentDetails {...generateProps() as any} />);
+    await TestUtils.flushPromises();
+
+    tree.update();
+
+    tree.find('#manageExperimentRecurringRunsBtn').at(0).simulate('click');
+    await TestUtils.flushPromises();
+    expect(tree.state('recurringRunsManagerOpen')).toBe(true);
+
+    tree.find('#closeExperimentRecurringRunManagerBtn').at(0).simulate('click');
+    await TestUtils.flushPromises();
+    expect(tree.state('recurringRunsManagerOpen')).toBe(false);
+
+    tree.unmount();
+  });
+
+  it('refreshes the number of active recurring runs when the recurring run manager is closed', async () => {
+    await mockNJobs(1);
+    const tree = TestUtils.mountWithRouter(<ExperimentDetails {...generateProps() as any} />);
+    await TestUtils.flushPromises();
+
+    tree.update();
+
+    // Called when the page initially loads to display the number of active recurring runs
+    expect(listJobsSpy).toHaveBeenCalledTimes(1);
+
+    tree.find('#manageExperimentRecurringRunsBtn').at(0).simulate('click');
+    await TestUtils.flushPromises();
+    expect(tree.state('recurringRunsManagerOpen')).toBe(true);
+
+    // Called in the recurring run manager to list the recurring runs
+    expect(listJobsSpy).toHaveBeenCalledTimes(2);
+
+    tree.find('#closeExperimentRecurringRunManagerBtn').at(0).simulate('click');
+    await TestUtils.flushPromises();
+    expect(tree.state('recurringRunsManagerOpen')).toBe(false);
+
+    // Called a third time when the manager is closed to update the number of active recurring runs
+    expect(listJobsSpy).toHaveBeenCalledTimes(3);
+
+    tree.unmount();
+  });
+
+  it('clears the error banner on refresh', async () => {
+    TestUtils.makeErrorResponseOnce(getExperimentSpy, 'test error');
+
+    const tree = shallow(<ExperimentDetails {...generateProps()} />);
+    await TestUtils.flushPromises();
+
+    // Verify that error banner is being shown
+    expect(updateBannerSpy).toHaveBeenLastCalledWith(expect.objectContaining({ mode: 'error' }));
+
+    (tree.instance() as ExperimentDetails).refresh();
+
+    // Error banner should be cleared
+    expect(updateBannerSpy).toHaveBeenLastCalledWith({});
+
+    tree.unmount();
+  });
+
+  it('navigates to the compare runs page', async () => {
+    const runs = [
```

Comment:
Use `mockNRuns`?

Reply:
I wanted to use this so that it's explicit where the query params come from